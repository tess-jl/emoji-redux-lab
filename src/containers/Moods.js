import React from 'react';
import Controls from '../components/controls/Controls';
import Face from '../components/face/Face';
import { drinkCoffee, eatSnack, takeNap, study } from '../actions/taskActions';
import { useSelector, useDispatch } from 'react-redux';


const Moods = () => {
  const { coffees, snacks, naps, studies } = useSelector(state => state);

  const dispatch = useDispatch();

  const handleSelection = task => {
    dispatch(task);
  };

  return (
    <>
      <Controls>
        <button onClick={() => handleSelection(drinkCoffee())}>coffee - {coffees}</button>
        <button onClick={() => handleSelection(eatSnack())}>snacks - {snacks}</button>
        <button onClick={() => handleSelection(takeNap())}>naps - {naps}</button>
        <button onClick={() => handleSelection(study())}>studies - {studies}</button>
      </Controls>
      <Face /> 
    </>
  );
};

export default Moods; 
