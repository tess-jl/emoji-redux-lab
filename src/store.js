const { createStore } = require('redux'); 
import reducer from './reducers/moodsReducer';

export default createStore(
  reducer,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);
