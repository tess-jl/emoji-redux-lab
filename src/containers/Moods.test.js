import React from 'react';
import { shallow } from 'enzyme';
import Moods from './Moods';
import { Provider } from 'react-redux';
import store from '../store';

describe('Moods component', () => {
  it('renders Moods', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Moods />
      </Provider>
    ).dive();
    expect(wrapper).toMatchSnapshot();
  });
});

