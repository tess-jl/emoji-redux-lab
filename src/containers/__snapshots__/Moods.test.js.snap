// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Moods component renders Moods 1`] = `<Moods />`;
